### ISSUE TITLE
Adam::update panics on scalar/1×1 params; integration AdamOptimizer does no updates

### ISSUE THREAD

**Issue by mcoughlin** [cool-japan/scirs#98]
Calling the public optimizer API Adam::default(..).update]with small parameters (scalars or 1×1 arrays) panics in AdamOp::compute with an ndarray index out-of-bounds.

Using the integration-layer integration::optim::AdamOptimizer avoids the panic, but step is a placeholder and performs no parameter updates, so training is frozen.

I think it would be good to:
(i) Fix AdamOp::compute (and similar SGD op) to handle scalar/1×1 parameters without panicking.
(ii) Implement actual parameter updates in integration::optim::AdamOptimizer::step or document that it’s unimplemented.

See below for a reproducible example.

use scirs2_autograd as ag;
use ag::optimizers::adam::Adam;
use ag::tensor_ops::*;
use ag::prelude::*;
use scirs2_core::ndarray::arr2;

fn main() {
    let data = arr2(&[[1.3], [1.6], [0.9], [1.1], [1.4]]);
    let n = data.nrows();

    let mut env = ag::VariableEnvironment::new();
    let m_id = env.name("m").set(arr2(&[[0.0]]));      // shape [1,1]
    let log_s_id = env.name("log_s").set(arr2(&[[-1.0]])); // shape [1,1]
    let mut adam = Adam::default("adam_optimizer", vec![m_id, log_s_id], &mut env);

    for _ in 0..3 {
        env.run(|ctx| {
            let x = ctx.placeholder("x", &[n as isize, 1]);
            let m = ctx.variable_by_id(m_id);
            let log_s = ctx.variable_by_id(log_s_id);
            let eps = random_normal(&[n as isize, 1], 0.0, 1.0, ctx);
            let s = exp(log_s);
            let mu = reshape(m, &[n as isize, 1]) + reshape(s, &[n as isize, 1]) * eps;

            let loglik = -0.5 * reduce_sum(square(x - mu), &[0, 1], false);
            let logprior = -0.5 * reduce_sum(square(mu), &[0, 1], false);
            let entropy = reduce_sum(log_s + 0.5 * (2.0 * std::f64::consts::PI).ln(), &[0, 1], false);
            let elbo = loglik + logprior + entropy;
            let loss = &elbo * -1.0;

            let grads = grad(&[loss], &[m, log_s]);
            let feeder = ag::Feeder::new().push(x, data.view().into_dyn());

            // Panics here with ndarray index OOB
            adam.update(&[m, log_s], &grads, ctx, feeder);
        });
    }
}
